Gordon360/Services/LostAndFoundService.cs

```diff
@@ -0,0 +1,338 @@
+            foreach (MissingItemReportViewModel item in missingModelList)
+            {
+                if (item.recordID != null)
+                {
+                    actionsTakenList.Add(GetActionsTaken((int)item.recordID));
+                }
+            }
+
+            IEnumerator<IEnumerable<ActionsTakenViewModel>> actionsTakenEnumerator = actionsTakenList.GetEnumerator();
+
+            missingModelList = missingModelList.Select(x => { actionsTakenEnumerator.MoveNext(); x.adminActions = actionsTakenEnumerator.Current; return x; });
+            return missingModelList;
+        }
+
+        /// <summary>
+        /// Gets a Missing by id, only allowed if it belongs to the username, or the user is an admin
+        /// </summary>
+        /// <param name="id">The ID of the missing item</param>
+        /// <param name="username">The username of the person requesting the data</param>
+        /// <returns>A Missing Item Report object, or null if no item matches the id</returns>
+        public MissingItemReportViewModel? GetMissingItem(int id, string username)
+        {
+            MissingItemData report;
+            bool isAdmin = Authorization.AuthUtils.GetGroups(username).Contains(Enums.AuthGroup.LostAndFoundAdmin);
+            // If user is admin, simply get the report
+            if (isAdmin)
+            {
+                report = context.MissingItemData.FirstOrDefault(x => x.ID == id);
+            }
+            else
+            {
+                // Otherwise check if the report belongs to the requesting user
+                var idNum = context.ACCOUNT.FirstOrDefault(x => x.AD_Username == username).gordon_id;
+                report = context.MissingItemData.FirstOrDefault(x => x.ID == id && x.submitterID == idNum);
+            }
+            // Typecast the data into the viewmodel
+            MissingItemReportViewModel returnReport = (MissingItemReportViewModel)report;
+
+            // Get the list of public admin actions on this report, and add them to the report.
+            returnReport.adminActions = GetActionsTaken(id, true);
+            
+            return returnReport;
+        }
+
+        /// <summary>
+        /// Gets a list of Actions Taken by id
+        /// </summary>
+        /// <param name="id">The ID of the associated missing item report</param>
+        /// <param name="getPublicOnly">Oonly get actions marked as public.  Default false.</param>
+        /// <returns>An ActionsTaken, or null if no item matches the id</returns>
+        public IEnumerable<ActionsTakenViewModel> GetActionsTaken(int id, bool getPublicOnly = false)
+        {
+            IEnumerable<ActionsTaken> actionsList;
+            if (getPublicOnly)
+            {
+                actionsList = context.ActionsTaken.Where(x => x.missingID == id && x.isPublic);
+            }
+            else
+            {
+                actionsList = context.ActionsTaken.Where(x => x.missingID == id);
+            }
```

Comment:
You can simplify this a little bit because `IQueryable` lets you chain LINQ operations that all get compiled into one SQL query:
```suggestion
            IEnumerable<ActionsTaken> actionsList = context.ActionsTaken.Where(x => x.missingID == id);
            if (getPublicOnly)
            {
                actionsList = actionsList.Where(x => x.isPublic);
            }
```

Reply:
Cool, I used this in a couple places in API #1111